docs/current/742989-cookbook.md

```diff
@@ -625,6 +625,44 @@ Set the Hashicorp Vault URI, namespace, role and secret identifiers as host envi
 
 [Learn more](./guides/723462-use-secrets.md)
 
+
+## Error handling
+
+### Aborting with unhandled error
+
+The following code listing demonstrates how to abort the script with an unhandled error, which allows debugging with the stack trace of the crash during development.
+
+```go file=./cookbook/snippets/error-handling/unhandled/main.go
+```
+
+### Aborting gracefully
+
+The following code listing demonstrates how to abort the script without crashing.
+
+```go file=./cookbook/snippets/error-handling/aborting/main.go
+```
+
+### Exit code and unexpected errors
+
+The following code listing demonstrates how to handle a non-zero exit code from running a command, with several use cases:
+- Difference between “test failed” and “failed to test”
+- Handle a specific exit code value
+- Handle an exec error without checking for the exit code
```

Comment:
```suggestion
- Handle an error without checking for the exit code
```

Reply:
For clarity, the point of "exec" here was to distinguish from other types of errors that don't have an exit code. So when you do this:

```go
if err != nil {
    var e *dagger.ExecError
    if errors.As(err, &e) {
        // exec error, handled and not propagated
        return nil
    }
    // non-exec error, propagated
    return err
}
```
But removing "exec" doesn't make it inaccurate necessarily, so it's ok.

Reply:
Since in my next comment removing "exec" loses meaning, if you have another suggestion, keep this one consistent with it as well.

Reply:
See if mod makes it clearer?